WebKit::isTestingIPC is used from IPC::, making IPC:: depend on WebKit::

kkinnunen-apple opened the pull request:
#### a997bf6b7a87abfdadf4c940a4fdcce024b8eab9
<pre>
WebKit::isTestingIPC is used from IPC::, making IPC:: depend on WebKit::
<a href="https://bugs.webkit.org/show_bug.cgi?id=243778">https://bugs.webkit.org/show_bug.cgi?id=243778</a>
rdar://problem/98441549

Reviewed by Antti Koivisto.

Move WebKit::isTestingIPC implementation to IPC:: in IPCUtilities.h

* Source/WebKit/GPUProcess/graphics/RemoteGraphicsContextGLCocoa.cpp:
* Source/WebKit/GPUProcess/webrtc/LibWebRTCCodecsProxy.mm:
(WebKit::LibWebRTCCodecsProxy::createH264Decoder):
(WebKit::LibWebRTCCodecsProxy::createH265Decoder):
(WebKit::LibWebRTCCodecsProxy::createVP9Decoder):
(WebKit::LibWebRTCCodecsProxy::createEncoder):
* Source/WebKit/GPUProcess/webrtc/RemoteSampleBufferDisplayLayer.cpp:
* Source/WebKit/Platform/IPC/IPCUtilities.cpp: Added.
(IPC::isTestingIPC):
(IPC::startTestingIPC):
(IPC::stopTestingIPC):
* Source/WebKit/Platform/IPC/IPCUtilities.h: Added.
(IPC::isTestingIPC):
* Source/WebKit/Platform/IPC/cocoa/ConnectionCocoa.mm:
* Source/WebKit/Shared/IPCConnectionTester.cpp:
* Source/WebKit/Shared/IPCStreamTester.cpp:
* Source/WebKit/Shared/IPCTester.cpp:
(WebKit::IPCTester::startMessageTesting):
(WebKit::IPCTester::createStreamTester):
(WebKit::IPCTester::createConnectionTester):
(WebKit::isTestingIPC): Deleted.
* Source/WebKit/Shared/IPCTester.h:
(WebKit::isTestingIPC): Deleted.
* Source/WebKit/Shared/ThreadSafeObjectHeap.h:
* Source/WebKit/Sources.txt:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/253421@main">https://commits.webkit.org/253421@main</a>
</pre>

cdumez commented on Source/WebKit/GPUProcess/webrtc/RemoteSampleBufferDisplayLayer.cpp:
```diff
@@ -29,7 +29,7 @@
 
 #include "GPUConnectionToWebProcess.h"
-#include "IPCTester.h"
+#include "IPCUtilities.h"
```
Shouldn't the header say something about "Testing" in the name? IPCUtilities doesn't make it clear this is for testing.

kkinnunen-apple replied:
The initial idea was that most likely there won't be many testing related functions in addition to this. This would mean the "Testing" header would be a bit too specific.
However, there likely is some utilities that could be moved to the header. This would mean that having testing utilities in "Utilities" header would be appropriate.

kkinnunen-apple committed:
WebKit::isTestingIPC is used from IPC::, making IPC:: depend on WebKit::
https://bugs.webkit.org/show_bug.cgi?id=243778
rdar://problem/98441549

Reviewed by Antti Koivisto.

Move WebKit::isTestingIPC implementation to IPC:: in IPCUtilities.h

* Source/WebKit/GPUProcess/graphics/RemoteGraphicsContextGLCocoa.cpp:
* Source/WebKit/GPUProcess/webrtc/LibWebRTCCodecsProxy.mm:
(WebKit::LibWebRTCCodecsProxy::createH264Decoder):
(WebKit::LibWebRTCCodecsProxy::createH265Decoder):
(WebKit::LibWebRTCCodecsProxy::createVP9Decoder):
(WebKit::LibWebRTCCodecsProxy::createEncoder):
* Source/WebKit/GPUProcess/webrtc/RemoteSampleBufferDisplayLayer.cpp:
* Source/WebKit/Platform/IPC/IPCUtilities.cpp: Added.
(IPC::isTestingIPC):
(IPC::startTestingIPC):
(IPC::stopTestingIPC):
* Source/WebKit/Platform/IPC/IPCUtilities.h: Added.
(IPC::isTestingIPC):
* Source/WebKit/Platform/IPC/cocoa/ConnectionCocoa.mm:
* Source/WebKit/Shared/IPCConnectionTester.cpp:
* Source/WebKit/Shared/IPCStreamTester.cpp:
* Source/WebKit/Shared/IPCTester.cpp:
(WebKit::IPCTester::startMessageTesting):
(WebKit::IPCTester::createStreamTester):
(WebKit::IPCTester::createConnectionTester):
(WebKit::isTestingIPC): Deleted.
* Source/WebKit/Shared/IPCTester.h:
(WebKit::isTestingIPC): Deleted.
* Source/WebKit/Shared/ThreadSafeObjectHeap.h:
* Source/WebKit/Sources.txt:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/253421@main
